src/content/en/fundamentals/engage-and-retain/push-notifications/web-push-protocol.md

```diff
@@ -0,0 +1,613 @@
+
+Technically a push service can reduce the TTL of a push message if it wants. You can tell if
+this has happened by examining the TTL header in the response from a push service.
+
+##### Topic
+
+This is an **optional header**.
+
+Topics are strings that can be used to replace any pending notifications with
+new notifications if they have matching topic names.
+
+This is useful in scenarios where the user's device might have been offline during which you
+might have sent three messages and you may want the user to receive only the latest message
+when their device is eventually turned back on.
+
+##### Urgency
+
+This is an **optional header**.
+
+This can be used to indicate to the push service how important a message is to the user. This
+could be used by the push service to help conserve the battery life of a users device by only
+waking up for important messages when battery is low.
+
+The header value should a string value of "very-low" | "low" | "normal" | "high". The default
+value is "normal".
+
+    Urgency: <very-low | low | normal | high>
+
+### Everything Together
+
+If you have further questions about how this all works you can always see how libraries trigger
+push messages on [the web-push-libs org](https://github.com/web-push-libs).
+
+Once you have an encrypted payload, and the headers above, you just need to make a POST request
+to the `endpoint` in a `PushSubscription`.
+
+So what do we do with the response to this POST request?
+
+### Response from Push Service
+
+Once you've made a request to a Push Service, you need to check the status code
+of the response as that'll inform you as to whether the request was successful
+or not.
+
+<table>
+  <tr>
+    <th>Status Code</th>
+    <th>Description</th>
+  </tr>
+  <tr>
+    <td>201</td>
+    <td>Created. The request to send a push message was received and accepted.
+    </td>
+  </tr>
+  <tr>
+    <td>429</td>
+    <td>Too many requests. Meaning your application server has reached a rate
+    limit with a push service. The push service should include a 'Retry-After'
+    header to indicate how long before another request can be made.</td>
+  </tr>
+  <tr>
+    <td>400</td>
+    <td>Invalid request. This generally means one of your headers is invalid
+    or poorly formatted.</td>
+  </tr>
+  <tr>
+    <td>404</td>
+    <td>Not Found. This is an indication that the subscription is expired
+    and can't be used. In this case you should delete the PushSubscription
+    and wait for the client to resubscribe the user.</td>
+  </tr>
+  <tr>
+    <td>410</td>
+    <td>Gone. The subscription is no longer valid and should be removed
+    from back end. This can be reproduced by calling `unsubscribe()` on a
+    PushSubscription.</td>
+  </tr>
+  <tr>
+    <td>413</td>
+    <td>Payload size too large. The minimum size payload a push service must
+    support is <a
+href="https://tools.ietf.org/html/draft-ietf-webpush-protocol-10#section-7.2">4096 bytes</a>
+(or 4kb).</td>
```

Comment:
If minimum size is something developers need to know, this isn't the place to document it. Please remove.